server/src/controller/apis/blogs.js

```diff
@@ -1,18 +1,21 @@
 import { Router } from 'express';
 import authInstance from '../../configs/authentication';
-import { authenticateUserBlog, blogExists } from '../../middlewares/blogs';
+import { authenticateUserBlog, createBlogValid } from '../../middlewares/blogs';
 import blogsAPI from '../../services/blogs/api';
+import {
+  saveBlog, fetchBlog, updateBlog, deleteBlog,
+} from '../../services/blogs/service';
 
 const router = Router();
 
 router.get('/health', blogsAPI.health);
 
-router.get('/:id', blogsAPI.readBlog);
+router.get('/:id', fetchBlog, blogsAPI.readBlog);
 
-router.post('/', authInstance.isValidUser(), blogsAPI.createBlog);
+router.post('/', createBlogValid, authInstance.isValidUser(), saveBlog, blogsAPI.createBlog);
 
-router.put('/:id', authInstance.isValidUser(), blogExists, authenticateUserBlog, blogsAPI.updateBlog);
+router.put('/:id', authInstance.isValidUser(), fetchBlog, authenticateUserBlog, updateBlog, blogsAPI.updateBlog);
 
-router.delete('/:id', authInstance.isValidUser(), blogExists, authenticateUserBlog, blogsAPI.deleteBlog);
+router.delete('/:id', authInstance.isValidUser(), fetchBlog, authenticateUserBlog, deleteBlog, blogsAPI.deleteBlog);
 
```

Comment:
It seems that the delete will be happening in two places. `deleteBlog` and `blogsAPI.deleteBlog`